<a href="https://colab.research.google.com/github/Garaujott/Projeto_SirioLibanes_Covid_Kaggle/blob/main/ProjetoSirioLibanes_PreparacaoDados.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Preparação do Ambiente e Observações

In [ ]:
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import time

# from sklearn.dummy import DummyClassifier
# from sklearn.ensemble import RandomForestClassifier
# from sklearn.linear_model import LogisticRegression
# from sklearn.metrics import accuracy_score
# from sklearn.metrics import classification_report
# from sklearn.metrics import plot_confusion_matrix
# from sklearn.metrics import roc_auc_score
# from sklearn.model_selection import cross_validate
# from sklearn.model_selection import RepeatedStratifiedKFold
# from sklearn.model_selection import StratifiedKFold
# from sklearn.model_selection import train_test_split
# from sklearn.tree import DecisionTreeClassifier

# Importação dos dados

In [ ]:
# base presente no Kaggle (https://www.kaggle.com/S%C3%ADrio-Libanes/covid19)

uri = 'https://github.com/Garaujott/Projeto_SirioLibanes_Covid_Kaggle/blob/main/dados/Kaggle_Sirio_Libanes_ICU_Prediction.xlsx?raw=true'

covid = pd.read_excel(uri)
covid.head(10)

PATIENT_VISIT_IDENTIFIER  AGE_ABOVE65  ...    WINDOW  ICU
0                         0            1  ...       0-2    0
1                         0            1  ...       2-4    0
2                         0            1  ...       4-6    0
3                         0            1  ...      6-12    0
4                         0            1  ...  ABOVE_12    1
5                         1            1  ...       0-2    1
6                         1            1  ...       2-4    1
7                         1            1  ...       4-6    1
8                         1            1  ...      6-12    1
9                         1            1  ...  ABOVE_12    1

[10 rows x 231 columns]

# Análise Exploratória e Pré-processamento dos Dados

## Análise Preliminar

In [ ]:
covid.describe()

PATIENT_VISIT_IDENTIFIER  ...          ICU
count               1925.000000  ...  1925.000000
mean                 192.000000  ...     0.267532
std                  111.168431  ...     0.442787
min                    0.000000  ...     0.000000
25%                   96.000000  ...     0.000000
50%                  192.000000  ...     0.000000
75%                  288.000000  ...     1.000000
max                  384.000000  ...     1.000000

[8 rows x 229 columns]

Avaliando a estrutura da base: as linhas representam as visitas/avaliações dos pacientes em cada uma das 5 janelas possíveis

A variável *PATIENT_VISIT_IDENTIFIER* é o código de identificação do paciente, esse código está entre 0 e 384. Isso significa que temos 385 pacientes na base, que foram avaliados 5 vezes, resultando nas 1925 linhas da tabela.

In [ ]:
# a base tem 385 indivíduos, mas eles aparecem 5 vezes cada. Por isso, que temos mais de 385 observações.

print(f'A base tem {len(covid["PATIENT_VISIT_IDENTIFIER"].unique())} indivíduos.')

A base tem 385 indivíduos.


## Limpeza dos Dados

## Análise Exploratória